### PR TITLE
feat: support deploying apisix in daemonset way

### PR DESCRIPTION
**Pull request by PGDream**
apisix chart support `DaemonSet` or `Deployment` , at deamonset.yaml delete podAffinity and nodeSelector;
delete Daemonset podAntiAffinity attribute, because Daemonset clear POD scheduler to k8s worker node

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tao12345666333) [inline .idea/workspace.xml:1]: Please remove this file. Thanks

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

**Comment by tao12345666333** [comment]
cool! thanks！

File: .idea/workspace.xml (hunk 1 above)
Comment (PGDream) [reply .idea/workspace.xml:1]: o

Comment (PGDream) [reply .idea/workspace.xml:1]: can add at .gitignore?

File: charts/apisix/templates/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            timeoutSeconds: 1
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                - /bin/sh
+                - -c
+                - "sleep 30"
+          env:
+            - name: INTRANET_IP
```
Comment (Gallardot) [inline charts/apisix/templates/daemonset.yaml:102]: Does `DaemonSet` need this? `Deployment` doesn't have that.

File: charts/apisix/templates/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      affinity:
+      {{- with .Values.apisix.affinity }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.apisix.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+{{- end }}
+{{- end }}
```
Comment (Gallardot) [inline charts/apisix/templates/daemonset.yaml:201]: We may need a new line here

File: charts/apisix/templates/deployment.yaml [hunk 4]
```diff
@@ -209,3 +209,4 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
 {{- end }}
+{{- end }}
```
Comment (Gallardot) [inline charts/apisix/templates/deployment.yaml:212]: ditto

File: charts/apisix/templates/daemonset.yaml (hunk 3 above)
Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:201]: yes, i merge `.Values.apisix.enabled` condation

File: charts/apisix/templates/daemonset.yaml (hunk 2 above)
Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:102]: enable `skywalking` feature, diff instance.
i add `Deployment` env attriubte

File: charts/apisix/templates/deployment.yaml (hunk 4 above)
Comment (PGDream) [reply charts/apisix/templates/deployment.yaml:212]: thx y

File: charts/apisix/templates/daemonset.yaml (hunk 2 above)
Comment (tokers) [reply charts/apisix/templates/daemonset.yaml:101]: @PGDream Sorry, cannot understand what you expressed. Could you describe it more elaborately?

Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:101]: After enabling skywalking, if you use an APISIX cluster, you need to distinguish which instance the link information comes from

Comment (tokers) [reply charts/apisix/templates/daemonset.yaml:101]: But who uses this environment variable?

Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:101]: skywalking user,check doamin flow

Comment (Gallardot) [reply charts/apisix/templates/daemonset.yaml:101]: But I can't find any use of this environment variable on either `skywalking-logger` plugin or `skywalking` plugin.

Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:101]: apisix dashboard controller panel  can  skywalking plugin  config `INTRANET_IP` env var

Comment (tokers) [reply charts/apisix/templates/daemonset.yaml:101]: @PGDream Could you give us a link about this?

Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:101]: sorry, `INTRANET_IP` env with `Prometheus`
apisix config.yaml: 
```
  prometheus:
    export_addr:
      ip: "${{INTRANET_IP}}"
      port: 9092
```

Comment (Gallardot) [reply charts/apisix/templates/daemonset.yaml:101]: I don't think this environment variable is a necessary option.  If you need it in your scenario, you can add it after this [PR](https://github.com/apache/apisix-helm-chart/pull/232) is merged

Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml:101]: OK

Comment (PGDream) [reply charts/apisix/templates/daemonset.yaml]: @tokers  please review

File: .idea/workspace.xml (hunk 1 above)
Comment (tokers) [reply .idea/workspace.xml:1]: It' still there, tracked by git.

File: charts/apisix/templates/daemonset.yaml (hunk 3 above)
Comment (tokers) [reply charts/apisix/templates/daemonset.yaml:194]: Still miss the EOL symbol @PGDream

**Comment by Gallardot** [comment]
@PGDream 

How's it going?  LGTM, except for some of the modifications we mentioned above. Feel free to ask for help if you need it.

**Comment by tokers** [comment]
@PGDream There is a conflict should be resolved.

**Comment by PGDream** [comment]
> @PGDream There is a conflict should be resolved.

ok

**Comment by PGDream** [comment]
> @PGDream
> 
> How's it going? LGTM, except for some of the modifications we mentioned above. Feel free to ask for help if you need it.

can be merged, nothing to modify;

**Comment by Gallardot** [comment]
LGTM

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Pull request merged**